src/components/Dialog.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <teleport to="body">
+    <div
+      v-if="visible"
+      class="overlay"
+      role="dialog"
+      aria-modal="true"
+      :aria-labelledby="'dialog-title'"
+      :aria-describedby="'dialog-message'"
+      tabindex="-1"
+      ref="dialog"
+      @keydown.tab.prevent="handleTab"
+      @keydown.esc="cancel"
+    >
+      <div class="dialog">
+        <wra-header size="h3" id="dialog-title">
+          {{ title }}
+        </wra-header>
+        <p id="dialog-message">{{ message }}</p>
+        <div class="dialog-buttons">
+          <wra-button
+            ref="confirmBtn"
+            :backgroundColor="isDelete ? 'wra-red' : 'wra-green'"
+            @click="confirm"
+          >
+            {{ confirmText }}
+          </wra-button>
+          <wra-button ref="cancelBtn" @click="cancel">
+            {{ cancelText }}
+          </wra-button>
+        </div>
+      </div>
+    </div>
```

Comment:
The semantic HTML for dialog boxes is <dialog>

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog

```suggestion
    </dialog>
```

Reply:
Completed